### ISSUE TITLE
Future Entries get listed (maybe intended)

### ISSUE THREAD

**Issue by marcorieser** [mitydigital/statamic-sitemapamic#11]
### Bug description

I have a collection with the following configuration:
```yaml
date_behavior:
  past: public
  future: unlisted
```
The future entries get listed in the sitemap but that is not what I expected. I get what I want by using `private` instead of `unlisted` but then the urls are no longer accessible for people who know them. 

Maybe this behavior is intended but feels a little counter intuitive to me. Alternatively a config option would be nice.

### Steps to reproduce

Create a collection with the following config:
```yaml
date_behavior:
  past: public
  future: unlisted
```
Create an entry in the future and open the generated sitemap.

### Environment and versions

```yaml
Environment
Application Name: Battery Quick Check
Laravel Version: 9.46.0
PHP Version: 8.1.13
Composer Version: 2.4.4
Environment: local
Debug Mode: ENABLED
URL: statamic-bqc.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 4
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 3.3.63 PRO

Statamic Addons
jacksleight/statamic-miniset: 0.3.1
jonassiewertsen/statamic-livewire: 2.9.0
marcorieser/statamic-vitals: 0.0.5
mitydigital/sitemapamic: 2.3.4
```

**Comment by martyf** [comment]
Thanks @marcorieser we've fixed this up in 2.3.5 which should now only show 'public' entries.

Let me know how you go.

**Comment by marcorieser** [comment]
Thank you, will check it out.
